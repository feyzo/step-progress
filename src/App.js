import React, { Component } from 'react';
import styled from 'styled-components';
import StepProgress from './components/StepProgress';

const AppContainer = styled.section`
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 20%;
`;

class App extends Component {
  constructor(props) {
    super(props);

    this.onStepChange = this.onStepChange.bind(this);

    this.state = {
      steps: [{
        name: 'Design',
        complete: true
      }, {
        name: 'Build',
        complete: true
      }, {
        name: 'Launch',
        complete: false
      }]
    };
  }

  onStepChange(step) {
    console.log(step);
  }

  render() {
    return (
      <AppContainer>
        <StepProgress
          steps={ this.state.steps }
          onStepChange={ this.onStepChange }
        />
      </AppContainer>
    );
  }
}

export default App;
