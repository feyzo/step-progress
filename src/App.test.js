import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';

jest.mock('./components/StepProgress', () => 'StepProgress');

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

test('Renders a App component', () => {
  const component = renderer.create(
    <App />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
