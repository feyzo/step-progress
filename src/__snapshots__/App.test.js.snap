exports[`test Renders a App component 1`] = `
<section
  className="dvhzlc">
  <StepProgress
    onStepChange={[Function]}
    steps={
      Array [
        Object {
          "complete": false,
          "enabled": true,
          "name": "Design",
        },
        Object {
          "complete": false,
          "enabled": false,
          "name": "Build",
        },
        Object {
          "complete": false,
          "enabled": false,
          "name": "Launch",
        },
      ]
    } />
</section>
`;
